eZ/Publish/Core/FieldType/Image/ImageThumbnailStrategy.php

```diff
@@ -29,11 +37,13 @@ class ImageThumbnailStrategy implements FieldTypeBasedThumbnailStrategy
     public function __construct(
         string $fieldTypeIdentifier,
         VariationHandler $variationHandler,
-        string $variationName
+        string $variationName,
+        ?LoggerInterface $logger = null
     ) {
         $this->fieldTypeIdentifier = $fieldTypeIdentifier;
         $this->variationHandler = $variationHandler;
         $this->variationName = $variationName;
+        $this->logger = $logger ?: new NullLogger();
```

Comment:
CS
```suggestion
        $this->logger = $logger ?? new NullLogger();
```

Reply:
Why though? We know `$logger` is set at this point.

Reply:
Technically `??` is more accurate. 

`?:` is a shorthand for ternary operator, which operates on `true`/`false` values (and `null` is implicitly cast to `false`). 

`??` checks against `null` without type casting - we have only `LoggerInterface|null`.

Reply:
> Why though? We know `$logger` is set at this point.

`isset` (and - by extension - `??` operator) does not only check that variable "is set", but also that it is not `null`.